tests/Builder/FluentPipelineFactoryTest.php

```diff
@@ -0,0 +1,26 @@
+<?php
+
+declare(strict_types=1);
+
+namespace MongoDB\Tests\Builder;
+
+use MongoDB\Builder\Pipeline;
+use MongoDB\Builder\Query;
+use MongoDB\Builder\Stage\FluentFactory;
+use MongoDB\Builder\Type\Sort;
+use PHPUnit\Framework\TestCase;
+
+class FluentPipelineFactoryTest extends TestCase
+{
+    public function testFluentPipelineFactory(): void
+    {
+        $pipeline = (new FluentFactory())
+            ->match(x: Query::eq(1))
+            ->project(_id: false, x: true)
+            ->sort(x: Sort::Asc)
+            ->getPipeline();
+
+        $this->assertInstanceof(Pipeline::class, $pipeline);
+        $this->assertCount(3, $pipeline->getIterator());
```

Comment:
Can you also add an assertion to ensure that the correct stages are added? As is we're only checking that three stages are added, but not what is actually being added.
A possible alternative would be to duplicate the existing stage tests and creating them with the fluent builder instead of the static one, e.g.:
```php
public function testEqualityMatch(): void
{
    $pipeline = (new FluentFactory())
        ->match(author: 'dave'),
        ->getPipeline();

    $this->assertSamePipeline(Pipelines::MatchEqualityMatch, $pipeline);
}
```

Reply:
I don't think replicating all tests is interesting. This single test can cover all specific cases.